### PR TITLE
test: SpaceRoles in NSTemplateSets

### PR DESCRIPTION
**Pull request by xcoulon**
Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: pkg/test/nstemplateset/nstemplateset_assertion_test.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+		t.Run("fails when actual user is not expected", func(t *testing.T) {
+			// given
+			mockT := test.NewMockT()
+			// when
+			nstemplateset.AssertThatNSTemplateSet(mockT, test.MemberOperatorNs, "foo", cl).HasSpaceRoles(
+				nstemplateset.SpaceRole("base-admin-123456b", "admin1"), // admin2 is not expected
+				nstemplateset.SpaceRole("base-viewer-123456b", "viewer1"),
+			)
+			// then: all good
```
Comment (rajivnathan) [inline pkg/test/nstemplateset/nstemplateset_assertion_test.go:65]: I guess this comment can be removed for all the error scenarios?

Comment (xcoulon) [reply pkg/test/nstemplateset/nstemplateset_assertion_test.go:65]: may I ask why? the "all good" comment is to explain that no error or failure should have been reported by the assertion

File: pkg/test/nstemplateset/nstemplateset.go [hunk 2]
```diff
@@ -36,6 +64,31 @@ func WithReferencesFor(nstemplateTier *toolchainv1alpha1.NSTemplateTier) Option
...
+			apply(nstemplateTier, nstmplSet)
+		}
+	}
+}
+
+type TierOption func(*toolchainv1alpha1.NSTemplateTier, *toolchainv1alpha1.NSTemplateSet)
+
+func WithSpaceRole(role, username string) TierOption {
+	return func(nstemplateTier *toolchainv1alpha1.NSTemplateTier, nstmplSet *toolchainv1alpha1.NSTemplateSet) {
+		if tierSpaceRole, found := nstemplateTier.Spec.SpaceRoles[role]; found {
```
[the remark this answers is not in the document]
Comment (rajivnathan) [reply pkg/test/nstemplateset/nstemplateset.go:78]: If the role does not exist then this function is a no op so I don't think it makes sense to provide a role that would not be found?

[the remark this answers is not in the document]
Comment (xcoulon) [reply pkg/test/nstemplateset/nstemplateset.go:78]: yes, that's right. Which is why I said that the caller should be sure that the role exists. But maybe we can improve this func to make it less error-prone, indeed

File: pkg/test/nstemplateset/nstemplateset_assertion_test.go (hunk 1 above)
Comment (xcoulon) [reply pkg/test/nstemplateset/nstemplateset_assertion_test.go:65]: ahh, I see. I did not realize you were pointing to a case where an error is expected! 🤦‍♂️ 
let me fix that in a subsequent PR :)

Comment (xcoulon) [reply pkg/test/nstemplateset/nstemplateset_assertion_test.go:65]: see https://github.com/codeready-toolchain/toolchain-common/pull/259

**Pull request merged**